perf/benchmark.jl

```diff
@@ -20,30 +22,49 @@ function benchmark_driver!(f, x...; f_displayname=string(f))
     print("  Run TapedFunction (compiled):")
     @btime $ctf($(x)...)
     GC.gc()
+
+    print("  Run TapedTask: ")
+    x = (x[1:end-1]..., produce);
+    # show the number of produce calls inside `f`
+    tf = Libtask.TapedFunction(f, x...);
+    tt = TapedTask(tf, x...); 
+    c = 0; while consume(tt)!==nothing c+=1 end;
+    print("#produce=", c);
+    f_task = (tf, x) -> begin 
+        # tf = Libtask.TapedFunction(f, x...);
+        tt = TapedTask(tf, x...); 
+        while consume(tt)!==nothing
+        end; 
+    end
+    @btime $f_task($f, $x)
+    GC.gc()
 end
 
 ####################################################################
 
 
-function rosenbrock(x)
+function rosenbrock(x, callback=nothing)
     i = x[2:end]
     j = x[1:end-1]
-    return sum((1 .- j).^2 + 100*(i - j.^2).^2)
+    ret = sum((1 .- j).^2 + 100*(i - j.^2).^2)
+    if callback !== nothing; callback(ret); end
```

Comment:
This is not idiomatic Julia code.

```suggestion
    callback !== nothing && callback(ret)
```
or multiple lines